components/remote_settings/src/config.rs

```diff
@@ -36,7 +36,17 @@ pub enum RemoteSettingsServer {
 }
 
 impl RemoteSettingsServer {
-    pub fn url(&self) -> Result<Url> {
+    /// Get the [url::Url] for this server
+    #[error_support::handle_error(Error)]
+    pub fn url(&self) -> ApiResult<Url> {
+        self.get_url()
+    }
+
+    /// Internal version of `url()`.
+    ///
+    /// The difference is that it uses `Error` instead of `ApiError`.  This is what we need to use
+    /// inside the crate.
```

Comment:
This is one annoying part about splitting up the errors into internal vs external errors.  `url()` needs to return an `ApiResult` since it's pub and used by consumers (nimbus), but then `Client::new` can't use it since it returns a `Result<>` type.